Sources/Orbit/Components/HorizontalScroll.swift

```diff
@@ -23,301 +28,521 @@ public enum HorizontalScrollItemHeight {
-                .padding(.horizontal, horizontalPadding)
-                .padding(.vertical, verticalPadding)
+        scrollView
+            .background(scrollViewWidthReader)
+            .onPreferenceChange(ScrollViewWidthPreferenceKey.self) {
+                scrollViewWidth = $0
             }
+            .clipped()
+            .padding(-clippingPadding)
+    }
+
+    @ViewBuilder var scrollView: some View {
+        if snapping {
+            snappingScrollView
+        } else {
+            nativeScrollViewReader
+        }
+    }
+
+    @ViewBuilder var nativeScrollViewReader: some View {
+        if #available(iOS 14.0, *) {
+            ScrollViewReader { scroll in
+                nativeScrollView
+                    .onChange(of: scrollDestinationID) {
+                        scroll.scrollTo($0, anchor: .bottom)
+                    }
+            }
+        } else {
+            nativeScrollView
+        }
+    }
+
+    @ViewBuilder var nativeScrollView: some View {
+        SwiftUI.ScrollView(.horizontal, showsIndicators: false) {
+            scrollViewContent
         }
     }
 
+    @ViewBuilder var snappingScrollView: some View {
+        GeometryReader { geometry in
+            scrollViewContent
+                .disabled(scrollingInProgress || isAnimating)
+                .background(heightReader)
+                .offset(x: snappingOffset)
+                .contentShape(Rectangle())
+                .simultaneousGesture(
+                    SimultaneousGesture(scrollingGesture, scrollStoppingGesture)
+                )
+                .valueChanged(scrollDestinationID) {
+                    scrollToID(id: $0, geometry: geometry)
+                }
+                .onReceive(timer) { output in
+                    guard let lastDate else { return }
+                    let animatingTime = output.timeIntervalSince(lastDate)
+                    animateOffset(currentTime: animatingTime)
+                }
+        }
+        .onPreferenceChange(ContentSizePreferenceKey.self) {
+            contentSize = $0
+        }
+        .onPreferenceChange(IDPreferenceKey.self) {
+            idPreferences = $0
+        }
+        .frame(height: resolvedItemHeight ?? contentSize.height, alignment: .top)
+    }
+
+    @ViewBuilder var scrollViewContent: some View {
+        HStack(alignment: .top, spacing: spacing) {
+            content
+                .frame(width: resolvedItemWidth, alignment: .leading)
+                .frame(height: resolvedItemHeight, alignment: .top)
+        }
+        .padding(clippingPadding)
+        .fixedSize(horizontal: false, vertical: true)
+    }
+
+    @ViewBuilder var heightReader: some View {
+        GeometryReader { geometry in
+            Color.clear.preference(
+                key: ContentSizePreferenceKey.self,
+                value: geometry.size
+            )
+        }
+    }
+
+    @ViewBuilder var scrollViewWidthReader: some View {
```

Comment:
Why is one just `heightReader` and the other `scrollViewWidthReader`?

Reply:
Renamed to `contentHeightReader` and `availableWidthReader`